### PR TITLE
WIP => [AppShell] Create router-based <AppShell />, take 2

### PR DESCRIPTION
**Pull request by damassi**
Force PR: https://github.com/artsy/force/pull/4226

#### Notes
- Artwork Page
  - [x] View in room 
  - [x] Mediator stuff -- contact gallery button, login 
- General
  - [x] Ensure that `react-head` tags exist in pages like artist / artwork, so that meta props are dynamically updated

shell

### CODE REVIEW

File: src/Artsy/Router/Utils/RenderStatus.tsx [hunk 1]
```diff
@@ -24,6 +24,7 @@ export const RenderPending: React.FC = props => {
           position: "fixed",
           left: 0,
           top: -6,
+          zIndex: 1000,
```
Comment (damassi) [inline src/Artsy/Router/Utils/RenderStatus.tsx]: FIXME

File: src/Artsy/Router/makeAppRoutes.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+        /**
+         * Intercept <a> tags on page and if contained within router route
+         * manifest, navigate via router versus doing a hard jump between pages.
+         */
+        catchLinks(window, href => {
+          const url = ROUTE_NAMESPACE + href
+          const foundUrl = props.router.matcher.matchRoutes(routes, url)
+
+          console.log("here!@")
```
Comment (damassi) [inline src/Artsy/Router/makeAppRoutes.tsx]: fixme

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@artsy/reaction",
-  "version": "21.0.5",
+  "version": "21.0.5-beta.0",
```
Comment (damassi) [inline package.json]: FIXME

File: src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkImageBrowser.tsx [hunk 4]
```diff
@@ -49,7 +49,7 @@ export class LargeArtworkImageBrowser extends React.Component<
     }
 
     return (
-      <Container>
+      <Container key={Math.random()}>
```
Comment (damassi) [inline src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkImageBrowser.tsx]: Safety mechanism to get around mount / unmount of flickity carousel, as commented on below. I don't have a good solution yet after a day of debugging. But setting a new key, the dom node reinitializes, getting around the problem. It's not noticeable in the app.

File: src/Apps/Search/Components/NavigationTabs.tsx [hunk 5]
```diff
@@ -120,7 +124,10 @@ export class NavigationTabs extends React.Component<Props> {
   render() {
     return (
       <Flex mx={[-2, 0]}>
-        <TabCarousel tabs={this.tabs()} />
+        <TabCarousel
+          key={`tab-carousel-${this.props.term}`}
```
Comment (damassi) [inline src/Apps/Search/Components/NavigationTabs.tsx]: Setting a key, ensure that flickity doesn't freak out react when unmounting page. (Only happens occasionally)

**Comment by mzikherman** [comment]
Superseded in https://github.com/artsy/reaction/pull/3043/files

**Pull request closed without merging**